File: examples/javascript/test/interactions/navigation.spec.js
const {Builder } = require('selenium-webdriver');
const assert = require("node:assert");

describe('Interactions - Navigation', function () {
    let driver;

    before(async function () {
        driver = new Builder()
          .forBrowser('chrome')
          .build();
    });

    after(async () => await driver.quit());

    it('Browser navigation test', async function () {
        //Convenient
        await driver.get('https://www.selenium.dev');

        //Longer way
        await driver.navigate().to("https://www.selenium.dev/selenium/web/index.html");
        let title = await driver.getTitle();
        assert.equal(title, "Index of Available Pages");

        //Back
        await driver.navigate().back();
        title = await driver.getTitle();
        assert.equal(title, "Selenium");

        //Forward
        await driver.navigate().forward();
        title = await driver.getTitle();
        assert.equal(title, "Index of Available Pages");

        //Refresh
        await driver.navigate().refresh();
        title = await driver.getTitle();
        assert.equal(title, "Index of Available Pages");
    });
});

File: examples/ruby/spec/interactions/navigation_spec.rb
require 'spec_helper'

RSpec.describe 'Browser' do
  let(:driver) { start_session }

  it 'navigates to a page' do
    driver.navigate.to 'https://www.selenium.dev/'
    driver.get 'https://www.selenium.dev/'
    expect(driver.current_url).to eq 'https://www.selenium.dev/'
  end

  it 'navigates back' do
    driver.navigate.to 'https://www.selenium.dev/'
    driver.navigate.to 'https://www.selenium.dev/selenium/web/inputs.html'
    driver.navigate.back
    expect(driver.current_url).to eq 'https://www.selenium.dev/'
  end

  it 'navigates forward' do
    driver.navigate.to 'https://www.selenium.dev/'
    driver.navigate.to 'https://www.selenium.dev/selenium/web/inputs.html'
    driver.navigate.back
    driver.navigate.forward
    expect(driver.current_url).to eq 'https://www.selenium.dev/selenium/web/inputs.html'
  end

  it 'refreshes the page' do
    driver.navigate.to 'https://www.selenium.dev/'
    driver.navigate.refresh
    expect(driver.current_url).to eq 'https://www.selenium.dev/'
  end
end
